scrunch/datasets.py

```diff
@@ -1441,12 +1481,34 @@ class Variable(object):
 
     CATEGORICAL_TYPES = {'categorical', 'multiple_response', 'categorical_array'}
 
-    def __init__(self, resource, dataset_resource):
-        self.resource = resource
-        self.url = self.resource.self
-        self.dataset = Dataset(dataset_resource)
+    def __init__(self, var_shoji, dataset):
+        self.shoji_tuple = var_shoji
+        self.is_instance = False
+        self._resource = None
+        self.url = None
+        # pass an Entity class to save time
+        if isinstance(dataset, pycrunch.shoji.Entity):
```

Comment:
Yes, most cases do pass a shoji entity, we are only passing shijo tuple when doing ds.values(). If we only want to support passing a Shijo Tuple then we would need to rewrite all code referring to Variable.